validator/sawtooth_validator/journal/completer.py

```diff
@@ -281,6 +282,8 @@ def add_batch(self, batch):
 
     def get_block(self, block_id):
         with self.lock:
+            if block_id == "HEAD":
```

Comment:
Overloading block_id here could be avoided by adding an explicit get_chain_head() or similar.